modules/assim.batch/R/autoburnin.R

```diff
@@ -0,0 +1,42 @@
+#' @name getBurnin
+#' @title Calculate burnin value
+#' @description Use `coda::gelman_plot` to calculate first iteration number where Gelman diagnostic never exceeds the specified threshold.
+#' @param jags_out List of MCMC sample matrices or `mcmc.list` object
+#' @param threshold Maximum value of Gelman diagnostic
+#' @examples
+#'      library(coda)
+#'      data(line)
+#'      burnin <- getBurnin(line, threshold = 1.05)
+#' @author Michael Dietze, Alexey Shiklomanov
+#' @export
+getBurnin <- function(jags_out, threshold = 1.1) {
+    library(coda)
+    if (!is.mcmc.list(jags_out)) jags_out <- makeMCMCList(jags_out)
+    png("/dev/null")
+    GBR <- gelman.plot(jags_out, autoburnin=FALSE)
+    dev.off()
+    burnin <- GBR$last.iter[tail(which(rowSums(GBR$shrink[,,2] > threshold) > 0), 1) + 1]
```

Comment:
I think you might need a `drop=FALSE` here as in `GBR$shrink[,,2,drop=FALSE] ` for the cases there is only one parameter